drivers/scmi-msg/scmi_pinctrl.c

```diff
@@ -0,0 +1,810 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2024, EPAM Systems
+ */
+#include <cdefs.h>
+#include <string.h>
+#include <errno.h>
+
+#include <drivers/pinctrl.h>
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+#include <lib/utils.h>
+#include <lib/utils_def.h>
+
+#include "common.h"
+#define PP(fmt, ...) \
+	INFO("==== %s %d " fmt "\n", __func__, __LINE__, ##__VA_ARGS__);
+
+#pragma weak pinctrl_get_groups_count
+#pragma weak pinctrl_get_functions_count
+#pragma weak pinctrl_get_pins_count
+#pragma weak pinctrl_get_pins
+#pragma weak pinctrl_get_function_name
+#pragma weak pinctrl_get_group_name
+#pragma weak pinctrl_get_group_pins
+#pragma weak pinctrl_get_function_groups
+#pragma weak pinctrl_request
+#pragma weak pinctrl_free
+
+uint16_t pinctrl_get_groups_count()
+{
+	return 0;
```

Comment:
I think here and below should be -ENOTSUP

Reply:
No get _count should return 0 as safe value.